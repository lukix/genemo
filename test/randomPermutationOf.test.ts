import rand from 'random-seed';
import Genemo from '../src';

describe('Find Sequence', () => {
  test('Sequence matches snapshot', () => {
    const { random } = rand.create('seed');
    const valuesSet = [-1, 0, 1, 2, 2, 3, 4, 5, 6, 7, 8, 9];

    const result = Genemo.randomPermutationOf(valuesSet)(random);
    expect(result).toMatchSnapshot('randomPermutationOf');
  });

  test('works for empty valuesSet', () => {
    const { random } = rand.create('seed');
    const valuesSet = [];

    const result = Genemo.randomPermutationOf(valuesSet)(random);
    expect(result).toStrictEqual([]);
  });
});
