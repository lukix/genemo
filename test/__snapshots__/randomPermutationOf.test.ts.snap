// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Find Sequence Sequence matches snapshot: randomPermutationOf 1`] = `
Array [
  2,
  0,
  8,
  6,
  3,
  1,
  -1,
  2,
  9,
  5,
  7,
  4,
]
`;
